failure_test.go

```diff
@@ -58,7 +58,7 @@ func TestCabooseFailures(t *testing.T) {
 		DoValidation:             false,
 		PoolWeightChangeDebounce: time.Duration(1),
 		PoolRefresh:              time.Millisecond * 50,
-		MaxRetrievalAttempts:     2,
+		MaxRetrievalAttempts:     3,
```

Comment:
This is necessary so that the one remaining good backend always gets used in line 144 at https://github.com/filecoin-saturn/caboose/blob/ebf9e25e4a94102072b215c4e7632da20fe74c15/failure_test.go#L142

Otherwise, it sometimes picks the two "invalid backends" for the key and test becomes flaky.